### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by George7h**
In this PR the following changes were implemented:

Add Redux Toolkit  npm install react-redux @reduxjs/toolkit.

Create a directory that will contain all your Redux logic (/src/redux)

Configure a Redux store (/src/redux/store.js)

Define a slice of state for books that:

Includes an array of books (initial state: empty array)
Includes a reducer that adds a book
Includes a reducer that removes a book
Define a slice of state for categories that:

Includes an array of categories (initial state: empty array)

Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

Structure your application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. Your files/folder structure could look like this:

/src
|---/components
|---/redux
|--- /books
| booksSlice.js
|--- /categories
| categoriesSlice.js
| store.js
| index.js

### CODE REVIEW

File: src/redux/categories/CategoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = 'Under construction';
```
Comment (Ibizugbe) [inline src/redux/categories/CategoriesSlice.js]: Nice job so far😉. Following the requirements in this milestone, the initialState is meant to be an empty array, similar to how you implemented it in the bookSlice😉. Kindly do well to fix that👍🏼.

image

File: src/redux/categories/CategoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+
+  },
```
Comment (Ibizugbe) [inline src/redux/categories/CategoriesSlice.js]: Just as required, the category reducer is meant to check status and always return "under construction". it's not meant to be empty as you have implemented😉. Kindly do well to fix that👍🏼.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @George7h 👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅

- [x] File structure follows the `duck pattern`👍🏼.
- [x] Linters are passing👍🏼.
- [x] The bookSlice has been created as required👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
```
Comment (zeff96) [inline src/redux/books/booksSlice.js]: - [ ] Hello! 😊 As pointed out by the previous reviewer, the initialState should have a state value for books that contains an empty array. 📚 This means that when the initial state is set up, the books property should start with no items in its array, creating a clean and empty slate for your application's book-related data. 📖✨

```sh
// ./src/redux/books/booksSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const booksSlice = createSlice({
  name: "books",
  initialState: {
    books: [],
  },
  reducers: {
    addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    },
    removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
  },
})

export const { addBook, removeBook } = booksSlice.actions

export default booksSlice.reducer

```

File: src/redux/categories/CategoriesSlice.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
```
Comment (zeff96) [inline src/redux/categories/CategoriesSlice.js]: - [ ] Hello! 😊 As pointed out by the previous reviewer, the initialState should have a state value for categories that contains an empty array. 📚 This means that when the initial state is set up, the categories property should start with no items in its array, creating a clean and empty slate for your application's category-related data. 📖✨

```sh
// ./src/redux/categories/categoriesSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})

export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer

```

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️
Hi @George7h , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost

 ## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: Correct GitFlow


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hello @George7h 👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ initial state has been updated correctly
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**